ArtOfIllusion/src/artofillusion/ScrollViewTool.java

```diff
@@ -0,0 +1,301 @@
+/* Copyright (C) 2017 by Petri Ihalainen
```

Comment:
>For a while I had the scroll events handled in the VC and MoveViewTool handle them a bit more to create the view cone graphics to the other views but obviously that was not a very logical way to do it.

That seems wrong -- A ViewerCanvas should not be forcibly drawn on if you are interacting with *different* ViewerCanvas. I'd almost wonder if a better option would be to temporarily add an object - an actual 3d wireframe - into the scene. It could be removed when the mouse action ends. This end of things requires more thought.

By click only, I mean that they should _only_ have Icons. They should be stripped of the "MetaTool" and "AltTool" arrangement. They would have the functions that they currently have in the 3.0.3 master branch. (In fact, they might even be removed entirely in the future.) The improved navigation tools that you are developing here would be an entirely separate implementation.

I would not worry about the size of a source file over-much. While it is true that a large class *can* be a sign of something wrong with a design, the real litmus test is whether the code makes sense in isolation. Would the proposed mouseHandler *ever* make sense outside the context of a ViewerCanvas? If not, it should remain in the ViewerCanvas, perhaps in a nested class to keep the class API cleaner. What to do with UVMeshViewer would take a little thought, but I don't expect it to be hard to block certain parts of the functionality.